### ISSUE TITLE
Thread titles with invalid characters in Windows export with MS-DOS like names

### ISSUE THREAD

**Issue by DannyDorito24** [Tyrrrz/DiscordChatExporter#1275]
### Version

2.43.3

### Flavor

CLI (Command-Line Interface)

### Platform

Ubuntu/MacOS

### Export format

JSON

### Steps to reproduce

Create threads with illegal filename characters (i.e. ? " / | : etc), export them, and threads containing those characters in their names will end up with 8 character names with a ~ character in them, much like old truncated MS-DOS names. (Example image attached)

### Details

I don't think this is a duplicate, but it's possible my search was incomplete as I'm not really sure how to describe the issue.

The most logical way to fix this, as I see it, would be to simply strip the illegal characters out of the filename. While not ideal, I imagine this is a limitation of .NET even on systems that support these characters (macOS and most Linux distros). Currently it is not ideal to browse through threads with names like this due to the fact that the truncated names don't seem to relate at all to what they were supposed to be named. 

Example from a recent export I did of a large server:
<img width="115" alt="Example of truncated names from an export" src="https://github.com/user-attachments/assets/6448c89e-7710-45a8-be90-dfa31504a29a">

Oddly, .NET does not throw any error or warning related to this, so the diagnosis here is of my own deduction. This has happened for many versions now, for me at least, so while I assume it's something in .NET, I'm somewhat confused as to why it hasn't been reported before if this is truly not a duplicate issue. Perhaps it's something else in the chain, but this happens on multiple systems.

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that this bug is reproducible on the latest version of the application
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [ ] I have sponsored this project
- [ ] I have not read any of the above and just checked all the boxes to submit the issue

**Comment by Tyrrrz** [comment]
Can you provide examples of the actual thread names?

**Comment by DannyDorito24** [comment]
```
"Where in the mc scource code are stronholds generated?" -> M3J8AK~K
"Can someone help me find the seed?" -> M4FRRX~4
"Can someone try find the NEW squid island world seed? Not the old one. The new one where they each" -> M6DBDK~9
"Help finding seed from MCPE 0.6.1 | Created between 3/4/13 & 3/5/13 (vids and world file)" -> M6J65P~G
<Thread name was a literal YouTube link> -> M7R5DI~A
```
And many more, all of them seem to have invalid characters for windows filenames. Also somewhat strange is the files have no extension when exported. Many of these threads are from a forum channel where anyone can open a request and so they aren't that important, but it's annoying and if this is wider spread than just me, I can see how it might be a bigger issue.

**Comment by Tyrrrz** [comment]
I was not able to reproduce this issue. I created a thread entitled `? " / | > Start` and the export resulted in a file named `DiscordChatExporter Test Server - thread-tests - _ _ _ _ _ Start [1304125458324918394].html` (illegal characters replaced by underscores).

It might be a Unix-specific issue. Can you upload the files that you experienced the issue with?

**Comment by DannyDorito24** [comment]
Hmm. I just tested it again with a single channel export, and it worked. I'll try to redo the export I guess? This was a large server so it will most likely take a few days to export again.

This is one of the files that exhibits the issue from an export I did in August. I added the file extension since macOS threw a fit at trying to zip it (guess certain implementations of zip don't like tilde?) and GitHub doesn't support extensionless files.
Below it is one I did today on my Mac. No idea how it fixed itself unless it's a macOS vs Ubuntu issue, which again I'll have to check here in a bit.
[M7R5DI~A.json](https://github.com/user-attachments/files/17668046/M7R5DI.A.json)
[Bot asset server - general - This Thread has characters that Windows can't handle ?|_\}{+_":';._,.)(!*@#^&% [1304157075600838746].json](https://github.com/user-attachments/files/17668084/Bot.asset.server.-.general.-.This.Thread.has.characters.that.Windows.can.t.handle._.%2B_._.%40.1304157075600838746.json)

**Comment by Tyrrrz** [comment]
@DannyDorito24 did it work on your second attempt?

**Comment by DannyDorito24** [comment]
It just now finished, and unfortunately, it didn't. Something I forgot to include is that when exporting an entire server, it throws an error for forum channels, that they have to be exported separately:
<img width="658" alt="Screenshot 2024-11-15 at 2 43 43 PM" src="https://github.com/user-attachments/assets/c552c9bd-82a5-4d85-8213-ccfd41236909">
However, these channels are present in the export and ones that don't have the presumably unsupported characters are perfectly fine, which is interesting.

**Comment by Tyrrrz** [comment]
The forum channel errors are unrelated and can be safely ignored if you're doing `exportall` or `exportguild`.

Are you still getting truncated thread names?

**Comment by DannyDorito24** [comment]
Yes, I've attached a few of them if it helps, but there's nothing obvious in the files to me other than the names.
[Truncated Names.zip](https://github.com/user-attachments/files/17805144/Truncated.Names.zip)

**Comment by Tyrrrz** [comment]
Hmm, can you share a server invite link? I can't reproduce it with threads on my own

**Comment by Tyrrrz** [comment]
Sorry @DannyDorito24, I've only found time to look into this now. Can you give me a link to the exact message that starts a thread with the problematic title? That way I can debug more quickly

**Comment by DannyDorito24** [comment]
https://discord.com/channels/720723932738486323/1290429497774051389/1290429497774051389
https://discord.com/channels/720723932738486323/1295056299935137872/1295056299935137872
https://discord.com/channels/720723932738486323/1291311397606526997/1291311397606526997

**Comment by Tyrrrz** [comment]
I tried to export those threads using the CLI (`export -c 1295056299935137872 -c 1291311397606526997 -t ...`) and the output filenames seem fine.


Windows:

![Image](https://github.com/user-attachments/assets/40cd5a2a-a009-4454-9e29-3e248bf16ebb)

Ubuntu:

![Image](https://github.com/user-attachments/assets/b1ffc534-0250-46d4-a367-2211f1e7d2f1)

Can't test it on macOS unfortunately.

**Comment by DannyDorito24** [comment]
Somehow, it is now fixed on macOS and ubuntu. Not sure how, but it works! Thanks!
